File: src/java/org/elastos/hive/Backup.java
package org.elastos.hive;

import org.elastos.hive.backup.ServiceBuilder;
import org.elastos.hive.service.PromotionService;

public class Backup extends ServiceEndpoint {
	private PromotionService promotionService;

	public Backup(AppContext context, String providerAddress) {
		super(context, providerAddress);
		this.promotionService = new ServiceBuilder(this).createPromotionService();
	}

	public PromotionService getPromotionService() {
		return this.promotionService;
	}
}

File: src/backup.ts
import {AppContext} from "./connection/auth/appcontext";
import {PromotionService} from "./service/promotion/promotionservice";
import {ServiceEndpoint} from "./connection/serviceendpoint";

/**
 * Represents the backup node.
 */
export class Backup extends ServiceEndpoint {
    private readonly promotionService: PromotionService;

    constructor(context: AppContext, providerAddress?: string) {
        super(context, providerAddress);
        this.promotionService = new PromotionService(this);
    }

    /**
     * Get the promotion service.
     */
    getPromotionService(): PromotionService {
        return this.promotionService;
    }
}
